packages/graphql-client/src/graphqlClient.ts

```diff
@@ -0,0 +1,143 @@
+
+export class GraphqlClient implements IChainDataClient<BoxWhere> {
+  private _client: Client;
+
+  constructor() {
+    this._client = createClient({
+      url: "TODO",
+      exchanges: [fetchExchange],
+      requestPolicy: "network-only"
+    });
+  }
+  async getUnspentBoxes(q: BoxQuery<ExtendedBoxWhere>): Promise<ChainClientBox[]> {
+    const query = gql<{ boxes: Omit<ChainClientBox, "confirmed">[] }, QueryBoxesArgs>`
+      query boxes(
+        $boxId: String
+        $boxIds: [String!]
+        $contract: String
+        $contracts: [String!]
+        $template: String
+        $tokenId: String
+        $skip: Int
+        $take: Int
+      ) {
+        boxes(
+          boxId: $boxId
+          boxIds: $boxIds
+          ergoTree: $contract
+          ergoTrees: $contracts
+          ergoTreeTemplateHash: $template
+          tokenId: $tokenId
+          skip: $skip
+          take: $take
+        ) {
+          boxId
+          transactionId
+          index
+          value
+          creationHeight
+          ergoTree
+          assets {
+            tokenId
+            amount
+          }
+          additionalRegisters
+        }
+      }
+    `;
+
+    const response = await this._client.query(query, {
+      boxId: q.where.boxId,
+      boxIds: q.where.boxIds,
+      address: q.where.contract,
+      addresses: q.where.contracts,
+      ergoTreeTemplateHash: q.where.template,
+      tokenId: q.where.tokenId,
+      skip: q.skip,
+      take: q.take,
+      spent: false
+    });
+
+    // What to do about confirmed?
+    return response.data?.boxes.map((box) => ({ ...box, confirmed: true })) ?? [];
+  }
+  async getLastHeaders(count: number): Promise<BlockHeader[]> {
+    const query = gql<{ blockHeaders: Header[] }, { count: number }>`
+      query blockHeaders($count: Int) {
+        blockHeaders(take: $count) {
+          headerId
+          timestamp
+          version
+          adProofsRoot
+          stateRoot
+          transactionsRoot
+          nBits
+          extensionHash
+          powSolutions
+          height
+          difficulty
+          parentId
+          votes
+          extension {
+            headerId
+          }
+          adProof {
+            headerId
+          }
+        }
+      }
+    `;
+
+    const response = await this._client.query(query, { count });
+
+    // What is size? Is it the block size?
+    // What is `transactionsId`?
+    return (
+      response.data?.blockHeaders.map((header) => ({
+        ...header,
+        id: header.headerId,
+        timestamp: Number(header.timestamp),
+        nBits: Number(header.nBits),
+        votes: header.votes.join(""),
+        size: 0,
+        transactionsId: "",
+        adProofsId: header.adProof.headerId,
+        extensionId: header.extension.headerId
+      })) ?? []
+    );
+  }
+  checkTransaction(transaction: SignedTransaction): Promise<boolean> {
+    // Size is missing from SignedTransaction! What to do?
```

Comment:
Size is not a required field, if required by GraphQL you can temporarily set it to `0` and open an issue on `ergo-graphql` repo so we can fix that later.

Reply:
Oh, you're right! It's optional so we can ignore it.